### PR TITLE
update create book

### PR DESCRIPTION
**Pull request by nghiabk**

### CODE REVIEW

**Comment by HungSonSN** [comment]
ok

File: app/views/admin/books/edit.html.erb [hunk 1]
```diff
@@ -3,21 +3,29 @@
...
       <%= f.label "Number Pages" %>
       <%= f.number_field :number_page, in: 1...1000, class: "form-control" %>
-      <%= image_tag @book.image.url if @book.image? %><br>
-      <%= f.label "Choose Image" %>
-      <%= f.file_field :image, accept: "image/jpeg,image/gif,image/png", 
-                               id: "upload" %>
-      <%= f.submit class: "btn btn-primary" %>
+
+      <%= f.fields_for :photos do |builder| %>
+        <%= render "photo_fields", :f => builder %>
```
Comment (dieunb) [inline app/views/admin/books/edit.html.erb]: conding convention

File: app/views/admin/books/new.html.erb [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
   <div class="col-md-6 col-md-offset-3">
-    <div class="show-result well">Book is created!</div>
-    <%= render "form" %>
-  </div>
-</div>
+    <%= form_for [:admin, @book] do |f| %>
+    <%= render "shared/error_messages", object: f.object %>
+    
+    <%= f.label "Category" %>
+    <%= f.select(:category_id, options_from_collection_for_select(Category.all,"id","name"), class: "form-control") %>
```
Comment (dieunb) [inline app/views/admin/books/new.html.erb]: missing a space after `Category.all`

File: app/views/photos/_photo.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<% if photo.important? %>
+<%= image_tag photo.image.url if photo.image? %>
```
Comment (dieunb) [inline app/views/photos/_photo.html.erb]: indent

File: app/models/book.rb [hunk 4]
```diff
@@ -12,8 +12,12 @@ class Book < ActiveRecord::Base
   validates :author, presence: true, length: {minimum: 6}
   validates :publish_date, :number_page, :rating, presence: true
   validate :image_size
+  validate :has_important_photo
 
   mount_uploader :image, ImageUploader
+
+  accepts_nested_attributes_for :photos, 
+  :reject_if => lambda {|a| a[:image].blank?}, allow_destroy: true
```
Comment (dieunb) [inline app/models/book.rb]:
```
reject_if:
```

**Pull request merged**